apis/kueue/v1beta2/multikueue_types.go

```diff
@@ -89,13 +93,16 @@ type MultiKueueClusterStatus struct {
 type MultiKueueCluster struct {
 	metav1.TypeMeta `json:",inline"`
 	// metadata is the metadata of the MultiKueueCluster.
+	// +optional
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
 	// spec is the specification of the MultiKueueCluster.
-	Spec MultiKueueClusterSpec `json:"spec,omitempty"`
+	// +optional
+	Spec MultiKueueClusterSpec `json:"spec,omitempty,omitzero"` //nolint:kubeapilinter // spec should not be a pointer
```

Comment:
Why adding omitzero here, but not for ClusterQueueSpec?

Reply:
I think its because we have optional fields for ClusterQueueSpec while from MultiKueueClusterSpec we have a required field in the spec.